### PR TITLE
[W09H03] cleanup DGTest tests

### PR DESCRIPTION
**Pull request by felixhauptmann**
* fix potential bug (unnecessarily instantiating `VideoContainer` could lead to one frame getting skipped)
* rearrange debug output and assertions (debug output would not get saved on failure)
* general cleanup

### CODE REVIEW

File: w09h03/test/pgdp/filter/DGTest.java [hunk 1]
```diff
@@ -375,9 +378,6 @@ public void test20() throws IOException {
                 Assertions.assertEquals(Color.YELLOW.getRGB(), output.getPixels().getRGB(i, j));
             }
         }
-
-        outputFrame("test20Output.png", output, "PNG");
-        outputFrame("test20Original.png", frame, "PNG");
```
Comment (Anatoly03) [inline w09h03/test/pgdp/filter/DGTest.java:381]: Do I understand correctly that this simply won't draw frames if tests fail?

File: w09h03/test/pgdp/filter/DGTest.java [hunk 2]
```diff
@@ -32,8 +32,12 @@ public class DGTest {
 
     private static void outputFrame(String pathname, Frame output, String PNG) throws IOException {
         if (!outputFiles) return;
-        File outputfile = new File(pathname);
-        ImageIO.write(output.getPixels(), PNG, outputfile);
+
+        File outputfile = new File("DGTest_output/" + pathname);
```
Comment (Anatoly03) [inline w09h03/test/pgdp/filter/DGTest.java:36]: Könnte vielleicht `output/` heißen

**Review by Anatoly03 — APPROVED** [review]
Remember to stay hydrated. <3

File: w09h03/test/pgdp/filter/DGTest.java (hunk 2 above)
Comment (felixhauptmann) [reply w09h03/test/pgdp/filter/DGTest.java:36]: war eine bewusste entscheidung um konflikte zu vermwiden

File: w09h03/test/pgdp/filter/DGTest.java [hunk 3]
```diff
@@ -43,10 +43,8 @@ private static void outputFrame(String pathname, Frame output, String PNG) throw
     @DisplayName("Test DG 01 Decode Noot first Frame")
     public void test01() throws IOException, IllegalVideoFormatException {
         // read the file
-        VideoContainer in;
         FrameProvider fp = new FrameProvider("noot.mp4");
-        in = new VideoContainer(fp);
```
[the remark this answers is not in the document]
Comment (felixhauptmann) [reply w09h03/test/pgdp/filter/DGTest.java:48]: 👍

File: w09h03/test/pgdp/filter/DGTest.java (hunk 1 above)
Comment (felixhauptmann) [reply w09h03/test/pgdp/filter/DGTest.java:381]: Frames should get saved in line 366

**Comment by felixhauptmann** [comment]
> Remember to stay hydrated. <3

Don't worry

File: w09h03/test/pgdp/filter/DGTest.java (hunk 2 above)
Comment (Anatoly03) [reply w09h03/test/pgdp/filter/DGTest.java:36]: Okay, danke! 👍

**Review by Anatoly03 — APPROVED** [review]

**Comment by felixhauptmann** [comment]
@nilsreichardt ready for merge

**Review by MaximilianAnzinger — COMMENTED** [review]
[🐧](https://emojipedia.org/penguin/)

**Pull request merged**